online-shop/src/app/header/header.component.html

```diff
@@ -1,7 +1,4 @@
-<div class="header">
+<div class="container">
   <h1>My online shop</h1>
-  <button [routerLink]="['/products']" >Product List</button>
+  <button (click)="products()">Products</button>
```

Comment:
Function and variable names should be meaningful. In this case when you click the button you navigate to Products page.
So you could have the following: navigateToProducts/navigateToProductsPage. 
Or in case you do multiple navigations in the same component. Imagine that you have Products, Rentals, and Refurbished. 
Here you don't want to have 3 functions->
navigateToProducts(){
  this.router.navigate(['/products'])
}
navigateToRentals(){
  this.router.navigate(['/rentals'])
}
navigateToRefurbished(){
  this.router.navigate(['/refurbished']).
}
Here another concept comes into play: REUSABILITY.
You can observe from those 3 functions that the only difference is the string value that is the .navigate parameter.
So here the best thing to do is to create a new function like this one->

handleNavigation(route:string){ //you can add optional params here like the id or other options...
   this.router.navigate([`${route}`]). //<--and here adapt as your situation requires
}

*as a side note you can keep the routes in an enum and in the router module, use those enum values to build the routes. It will help in case you will refactor the routing, or change some paths.